docs/manual/guides/formular-speichern.de.md

```diff
@@ -0,0 +1,550 @@
+übereinstimmen.
+{{% /notice %}}
+
+
+### Der Hook prepareFormData
+
+Die Datenbanktabelle ``tl_calendar_events`` beinhaltet zahlreiche Felder. Nicht alle sind für unser Beispiel notwendig.
+Wir benötigen zumindest Einträge für die Felder »pid«, »title«, »alias«, »author«, »startDate«, »startTime«, »endDate«, 
+»endTime« und »published«, damit wir im Contao Backend eine gültige Darstellung unserer Reservierungen erhalten. Optional 
+wären Einträge für »location«, »description« oder »teaser«. 
+
+Unser Formular bedient zur Zeit zumindest die Felder »title«, »startDate« und optionales »endDate«. Die weiteren Felder
+könnte man über versteckte Formularfelder hinzufügen. Diese Vorgehensweise ermöglicht jedoch die Manipulation der zu 
+übermittelten Formulardaten. Aus diesem Grund verwenden wir den Hook 
+»[prepareFormData](https://docs.contao.org/dev/reference/hooks/prepareFormData/)«.
+
+
+Dieser Hook greift nach der Formular Übermittlung aber vor der eigentlichen Datenspeicherung. Du kannst darüber 
+die übermittelten Daten bearbeiten und neue Informationen hinzufügen. Erstelle dir in deiner Contao Installation 
+die Verzeichnisse ``src\EventListener\`` und darin die Datei ``PrepareFormDataListener.php`` mit folgendem Inhalt:
+
+
+```php
+// src/EventListener/PrepareFormDataListener.php
+
+namespace App\EventListener;
+
+use Contao\CoreBundle\ServiceAnnotation\Hook;
+use Contao\Form;
+use Contao\CoreBundle\Slug\Slug;
+
+/**
+ * @Hook("prepareFormData")
+ */
+class PrepareFormDataListener 
+{
+  private $slug;
+
+  public function __construct(Slug $slug)
+  {
+      $this->slug = $slug;
+  }  
+
+  public function __invoke(array &$submittedData, array $labels, array $fields, Form $form): void
+  {
+    // set id of form, eventarchiv and author
+    $idForm         = 3;
+    $idEventArchiv  = 5;
+    $idAuthor       = 1;
+    
+    $ticketNr = strtotime("now");
+    
+    // form restriction 
+    if ($form->id == strval($idForm)) {
+
+      // mandatory fields
+      $submittedData['pid']       = $idEventArchiv;
+      $submittedData['author']    = $idAuthor;
+      $submittedData['published'] = 1;
+      
+      $tmpTitle = $submittedData['title'] . " Ticket: " . $ticketNr;
+      $submittedData['title'] = $tmpTitle;
+      $submittedData['alias'] = $this->slug->generate(
+        $tmpTitle, ['validChars' => 'a-z0-9', 'locale' => 'de', 'delimiter' => '-']
+      );      
+      
+      $submittedData['startTime'] = strtotime($submittedData['startDate']);
+      
+      // optional fields
+      if (!empty(trim($submittedData['endDate']))) {
+        $submittedData['endTime'] = strtotime($submittedData['endDate']);
+      } else {
+        $submittedData['endTime'] = null;
+        $submittedData['endDate'] = null;
+      }
+      
+      $submittedData['location'] = "Bootsverein Anleger";
+      $submittedData['teaser'] = "Reservierung Vereins-Boot: <br>" . $tmpTitle;
+
+    }
+  }
+}
+```
```

Comment:
Please use 4 spaces for indentation, as with the rest of the documentation :). Also you currently have a mix.

Reply:
> Also you currently have a mix.
What do you mean by that?

Reply:
Here you are using an indentation of 4 spaces: https://github.com/contao/docs/pull/781/files#diff-1861aa9946af452cb28b092eb180dbb39d03fb7bed158b3c570739275f08adcaR106
Everywhere else 2 spaces.

Reply:
Thanks - understood